AstSemantics.md

```diff
@@ -273,39 +275,53 @@ nested. This guarantees that all resulting control flow graphs are well-structur
 
 ## Calls
 
-Direct calls to a function specify the callee by index into a function table.
+Each function has a *signature*, which consists of:
 
-  * `call_direct`: call function directly
+  * Return types, which are a sequence of local types
+  * Argument types, which are a sequence of local types
+
+Note that WebAssembly itself does not support variable-length argument lists.
+C and C++ compilers are expected to implement this functionality by storing
+an array of arguments in a buffer in linear memory and passing a pointer to
+the buffer.
```

Comment:
I'd clarify: store a struct of naturally-aligned elements to the stack, where the struct's elements are the variable argument, ordered as they were from the callee, and pass an opaque pointer to the start of this struct. What's the type of this pointer? It's technically `void*` from a C perspective, but `byte*` or `uint8*` is more sensible for WebAssembly.

Link to: https://chromium.googlesource.com/native_client/pnacl-llvm/+/master/lib/Transforms/NaCl/ExpandVarArgs.cpp
?
I think it does exactly what we want :-)